### PR TITLE
ci: add local roachtest wrappers

### PR DESCRIPTION
**Pull request by rail**
This PR add wrappers to run FIPS local roachtests in CI.

Epic: DEVINF-478
Informs: #98824
Release note: None

### CODE REVIEW

File: build/teamcity/cockroach/ci/tests/local_roachtest_fips.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+dir="$(dirname $(dirname $(dirname $(dirname $(dirname "${0}")))))"
+
+source "$dir/teamcity-support.sh"  # For $root
```
Comment (healthy-pod) [inline build/teamcity/cockroach/ci/tests/local_roachtest_fips.sh:7]: nit: `$root` is not used though right?

File: build/teamcity/cockroach/ci/tests/local_roachtest_fips_impl.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -euox pipefail
+
+fips_enabled=$(cat /proc/sys/crypto/fips_enabled)
+
+if [[ $fips_enabled != "1" ]]; then
+  echo "FIPS mode is not enabled. Exiting."
+  exit 1
```
Comment (healthy-pod) [inline build/teamcity/cockroach/ci/tests/local_roachtest_fips_impl.sh:9]: Will this be added as a build config to `Bazel Essential CI`? Should we `exit 0` if FIPS is not enabled? i.e. noop

**Review by healthy-pod — COMMENTED** [review]
other than the two questions :lgtm:

File: build/teamcity/cockroach/ci/tests/local_roachtest_fips.sh (hunk 1 above)
Comment (rail) [reply build/teamcity/cockroach/ci/tests/local_roachtest_fips.sh:7]: It's implicitly used by `run_bazel`

File: build/teamcity/cockroach/ci/tests/local_roachtest_fips_impl.sh (hunk 2 above)
Comment (rail) [reply build/teamcity/cockroach/ci/tests/local_roachtest_fips_impl.sh:9]: The TeamCity wrapper calls this script only when it exists in the source tree. In other words it's called only when we need to call it.
This particular check happens after we ensure we need to run this test and checks that the underlying OS has proper support for FIPS.

Comment (healthy-pod) [reply build/teamcity/cockroach/ci/tests/local_roachtest_fips_impl.sh:9]: right right now I remember you talked about FIPS agents

**Review by healthy-pod — APPROVED** [review]

**Comment by rail** [comment]
FTR, it depends on https://github.com/cockroachdb/cockroach/pull/98824 (`run_bazel_fips`).

**Comment by rail** [comment]
TYFTR!

**Comment by rail** [comment]
bors r=healthy-pod

**Pull request merged**